backend/tournesol/serializers/entity.py

```diff
@@ -216,16 +215,28 @@ class Meta:
         ]
 
     def validate_uid(self, value):
-        split_uid = value.split(Entity.UID_DELIMITER)
+        """
+        Validate the `uid` against the regex provided by the entity class.
+        """
+        split_uid = value.split(UID_DELIMITER)
 
         if len(split_uid) <= 1 or not split_uid[1]:
             raise ValidationError("Malformed `uid`.")
 
+        poll = self.context.get("poll")
+        if poll:
```

Comment:
Currently all views using a `RelatedEntitySerializer` inherit from `PollScopedViewMixin` and thus the `poll` variable is never `None`.

I'm not sure about the relevance of this condition here. Are there future cases in which we will use the `RelatedEntitySerializer` in a view not related to a poll? I've opted for a safe approach but if we are confident that all operations on a related entity will be executed in the scope a of poll, we could remove this condition.

Reply:
Yes, if we consider that a poll is always defined in this context, it seems reasonable to write `self.context["poll"]`, and fail with a `KeyError` as soon as possible.